Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm

```diff
@@ -670,6 +681,22 @@ static bool isCopyDisplayedPixelBufferAvailable()
     m_videoLayerManager->setTextTrackRepresentationLayer(representationLayer);
 }
 
+bool MediaPlayerPrivateWebM::hasSingleSecurityOrigin() const
+{
+    if (m_resolvedOrigin && m_requestedOrigin)
+        return m_resolvedOrigin->isSameSchemeHostPort(*m_requestedOrigin);
+    return false;
+}
+
+std::optional<bool> MediaPlayerPrivateWebM::wouldTaintOrigin(const SecurityOrigin& origin) const
+{
+    for (auto& responseOrigin : m_origins) {
+        if (!origin.isSameOriginDomain(*responseOrigin))
```

Comment:
I don't think that's right.

This is checking that all redirects are the same origin at all point, this doesn't have to be so long that CORS passed.

So here we want to return early if
- hasSingleSecurityOrigin is true and we got no redirect.
- we passed earlier CORS along the way